File: server/tests/fixtures/alpha_rule_triggers/python/arch_data_model.py
# Should trigger: arch.data_model
from dataclasses import dataclass
from typing import Optional, List
from datetime import datetime

@dataclass
class User:
    """User data model - core domain entity"""
    id: int
    username: str
    email: str
    created_at: datetime
    is_active: bool = True
    profile_picture: Optional[str] = None

@dataclass
class Order:
    """Order data model - business entity"""
    order_id: str
    user_id: int
    items: List[str]
    total_amount: float

File: server/tests/fixtures/alpha_rule_triggers/typescript/arch_data_model.ts
// Should trigger: arch.data_model
// TypeScript interface data model
interface User {
    id: number;
    username: string;
    email: string;
    createdAt: Date;
    isActive: boolean;
    profilePicture?: string;
}

// TypeScript type with decorator pattern
interface Order {
    orderId: string;
    userId: number;
    items: string[];
    totalAmount: number;
}

// Class-based data model
class Product {
    constructor(
        public id: number,
        public name: string,
        public price: number,
        public category: string
    ) {}
}
